earthfile2llb/converter.go

```diff
@@ -2416,6 +2429,65 @@ func (c *Converter) internalRun(ctx context.Context, opts ConvertRunOpts) (pllb.
 	}
 }
 
+// detectAWSCredentials determines whether AWS envs are set and/or ~/.aws
+// exists. These are then used when fetching the secrets during a build.
+func (c *Converter) detectAWSCredentials(ctx context.Context) ([]llb.RunOption, []string, error) {
+
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to determine user home directory")
+	}
+
+	credsPath := fmt.Sprintf("%s/.aws/credentials", homeDir)
```

Comment:
maybe it's because it's still a work in progress, but I don't see where this file gets used.

Reply:
nevermind.... I scrolled down and saw the secret provider below -- this is just to check that these exist before hand.

in this case I think the majority of this error checking could be moved into a `util/...` package, then call it from the interrupter since it's related to the secondary-level parsing of flags, then only keep the llb-specific code in the converter source code.